app/src/main/java/tech/relaycorp/courier/domain/CollectPublicCargo.kt

```diff
@@ -0,0 +1,49 @@
+package tech.relaycorp.courier.domain
+
+import kotlinx.coroutines.flow.collect
+import tech.relaycorp.courier.data.database.StoredMessageDao
+import tech.relaycorp.courier.data.disk.DiskRepository
+import tech.relaycorp.courier.data.model.MessageAddress
+import tech.relaycorp.courier.data.model.MessageType
+import tech.relaycorp.courier.data.model.StoredMessage
+import tech.relaycorp.courier.data.network.CogRPC
+import javax.inject.Inject
+
+class CollectPublicCargo
```

Comment:
To be more consistent with the terminology used in the specs, I wouldn't use the term "public cargo" as it doesn't exist -- "public" or "private" only applies to nodes (gateways and endpoints).

Instead, I'd probably add the word "client" to the package name and name this class `CargoCollection`.